src/transformers/trainer.py

```diff
@@ -1009,7 +1013,17 @@ def create_optimizer(self):
                 },
             ]
 
-            optimizer_cls, optimizer_kwargs = Trainer.get_optimizer_cls_and_kwargs(self.args)
+            optimizer_cls, optimizer_kwargs = Trainer.get_optimizer_cls_and_kwargs(self.args, opt_model)
+
+            # Overwrite `params` in case it's created by `get_optimizer_cls_and_kwargs`
+            # e.g. for GaLoRe optimizer.
```

Comment:
Should the word `galore` be written as `Galore`, `GaLore`, or `GaLoRe`?

Reply:
+1 for `GaLore`, as defined in their [repo](https://github.com/jiaweizzhao/GaLore).

Reply:
done!